test/e2e/aks_webhooks.go

```diff
@@ -0,0 +1,74 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-05-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	infrav1exp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	expv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+)
+
+type AKSWebhooksSpecInput struct {
+	Cluster       *clusterv1.Cluster
+	MachinePools  []*expv1.MachinePool
+	WaitIntervals []interface{}
+}
+
+func AKSWebhooksSpec(ctx context.Context, inputGetter func() AKSWebhooksSpecInput) {
+	input := inputGetter()
+
+	settings, err := auth.GetSettingsFromEnvironment()
+	Expect(err).NotTo(HaveOccurred())
+	subscriptionID := settings.GetSubscriptionID()
+	auth, err := settings.GetAuthorizer()
+	Expect(err).NotTo(HaveOccurred())
+	agentpoolClient := containerservice.NewAgentPoolsClient(subscriptionID)
+	agentpoolClient.Authorizer = auth
+	mgmtClient := bootstrapClusterProxy.GetClient()
+	Expect(mgmtClient).NotTo(BeNil())
+
+	amc := &infrav1exp.AzureManagedCluster{}
+	err = mgmtClient.Get(ctx, types.NamespacedName{
+		Namespace: input.Cluster.Namespace,
+		Name:      input.Cluster.Name,
+	}, amc)
+
+	By("Validating that AzureManagedCluster webhooks are working")
+	By("Validating that we can't manually add an annotation to AzureManagedCluster with a prefix of 'infrastructure.cluster.x-k8s.io/custom-header-'")
+	amc.ObjectMeta.Annotations[azure.CustomHeaderPrefix+"foo"] = "bar"
+	err = mgmtClient.Update(ctx, amc)
+	Expect(err).To(HaveOccurred())
```

Comment:
I think it would be worth doing some sort of minimal "this is an error from the webhook" validation here to be safe, like [`apierrors.IsForbidden`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsForbidden) (assuming that's correct). I've definitely made mistakes when creating tests that generate HTTP 409 conflicts or similarly unexpected errors so I'd personally appreciate a little bit of guardrail here.